infrap4d/infrap4d_lite.cc

```diff
@@ -0,0 +1,32 @@
+// infrap4d_main without krnlmon
```

Comment:
File _infrap4d/infrap4d_main.cc_ also can cover the same functionality if we can manage the _krnlmon_ specific code under feature enablement.

If we follow the newer approach, it might become a maintenance issue latter stages.

Definitely there is no logic or functionality issue here. We need to look at the maintainability of the code.

Reply:
That was my original intention, until I discovered just how much code I was going to have to conditionalize.

Heavily conditionalized files are hard to maintain, too. It isn't worth it for this little code.
- _infrap4d_main.cc_ is 50 lines.
- _infrap4d_line.cc_ is 32 lines.

We can always combine them later, if the need arises.

-----

On the design front, there is a great deal to be said for providing a separate _main_ file for each infrap4d target configuration. It makes it easy to add and maintain new configurations (e.g., for SONiC), and to do things like defining target-specific command-line flags (or flags with different default values).

In addition to which, not having to deal with #ifdefs will make life a whole lot more pleasant when we move to Bazel.

We'll have a better idea of how to evolve when we have more targets.